stl/inc/format

```diff
@@ -2557,8 +2557,202 @@ template <_Format_supported_charT _CharT, class _Traits>
 struct formatter<basic_string_view<_CharT, _Traits>, _CharT>
     : _Formatter_base<basic_string_view<_CharT, _Traits>, _CharT, _Basic_format_arg_type::_String_type> {};
 
-using format_context  = basic_format_context<back_insert_iterator<string>, string::value_type>;
-using wformat_context = basic_format_context<back_insert_iterator<wstring>, wstring::value_type>;
+template <class _Ty>
+class _Fmt_buffer {
+private:
+    _Ty* _Ptr        = nullptr;
+    size_t _Size     = 0;
+    size_t _Capacity = 0;
+
+protected:
+    explicit _Fmt_buffer(const size_t _Sz) noexcept : _Size(_Sz), _Capacity(_Sz) {}
+    _Fmt_buffer(_Ty* _Data, const size_t _Sz, const size_t _Cap) noexcept : _Ptr(_Data), _Size(_Sz), _Capacity(_Cap) {}
+
+    void _Set(_Ty* _Buf_data, const size_t _Buf_capacity) noexcept {
+        _Ptr      = _Buf_data;
+        _Capacity = _Buf_capacity;
+    }
+
+    virtual void _Grow(size_t _Capacity) = 0;
+
+public:
+    using value_type      = _Ty;
+    using const_reference = const _Ty&;
+
+    _Fmt_buffer(const _Fmt_buffer&) = delete;
+    void operator=(const _Fmt_buffer&) = delete;
+
+    _NODISCARD _Ty* begin() noexcept {
+        return _Ptr;
+    }
+    _NODISCARD _Ty* end() noexcept {
+        return _Ptr + _Size;
+    }
+
+    _NODISCARD size_t size() const noexcept {
+        return _Size;
+    }
+
+    _NODISCARD size_t capacity() const noexcept {
+        return _Capacity;
+    }
+
+    void clear() noexcept {
+        _Size = 0;
+    }
+
+    void try_resize(const size_t _Count) {
+        try_reserve(_Count);
+        _Size = _Count <= _Capacity ? _Count : _Capacity;
+    }
+
+    void try_reserve(const size_t _New_capacity) {
+        if (_New_capacity > _Capacity) {
+            _Grow(_New_capacity);
+        }
+    }
+
+    void push_back(const _Ty _Value) {
+        try_reserve(_Size + 1);
+        _Ptr[_Size++] = _Value;
+    }
+};
+
+struct _Fmt_buffer_traits {
+    explicit _Fmt_buffer_traits(ptrdiff_t) {}
+    _NODISCARD size_t count() const noexcept {
+        return 0;
+    }
+    _NODISCARD size_t limit(const size_t _Size) noexcept {
+        return _Size;
+    }
+};
+
+class _Fmt_fixed_buffer_traits {
+private:
+    ptrdiff_t _Count = 0;
+    ptrdiff_t _Limit;
+
+public:
+    explicit _Fmt_fixed_buffer_traits(const ptrdiff_t _Lim) noexcept : _Limit(_Lim) {}
+    _NODISCARD size_t count() const noexcept {
+        return static_cast<size_t>(_Count);
+    }
+    _NODISCARD size_t limit(const size_t _Size) noexcept {
+        size_t _Avail = static_cast<size_t>(_Limit > _Count ? _Limit - _Count : 0);
+        _Count += _Size;
+        return _Size < _Avail ? _Size : _Avail;
+    }
+};
+
+inline constexpr size_t _Fmt_buffer_size = 256;
+
+template <class _OutputIt, class _Ty, class _Traits = _Fmt_buffer_traits>
+class _Fmt_iterator_buffer final : public _Traits, public _Fmt_buffer<_Ty> {
+private:
+    _OutputIt _Out;
+    _Ty _Data[_Fmt_buffer_size];
+
+protected:
+    void _Grow(size_t) final override {

```

Comment:
I don't know if there is convention regarding `final` and `override` but I couldn't find any function having both of them so I guess this should be just `final`. Ditto below

Reply:
We still have some vitual +override but i think it is planned to remove that. +1 for remiving override here

Reply:
Removed override.